zkevm-circuits/src/evm_circuit/execution/pop.rs

```diff
@@ -0,0 +1,170 @@
+use crate::{
+    evm_circuit::{
+        execution::{
+            bus_mapping_tmp::{Block, Call, ExecStep, Transaction},
+            ExecutionGadget,
+        },
+        step::ExecutionState,
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{
+                ConstraintBuilder, StepStateTransition, Transition::Delta,
+            },
+            Cell, Word,
+        },
+    },
+    util::Expr,
+};
+use bus_mapping::eth_types::ToLittleEndian;
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+#[derive(Clone, Debug)]
+pub(crate) struct PopGadget<F> {
+    same_context: SameContextGadget<F>,
+    value: Cell<F>,
+}
+
+impl<F: FieldExt> ExecutionGadget<F> for PopGadget<F> {
+    const NAME: &'static str = "POP";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::POP;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let value = cb.query_cell();
+
+        // Pop the value from the stack
+        cb.stack_pop(value.expr());
+
+        // State transition
+        let step_state_transition = StepStateTransition {
+            rw_counter: Delta(1.expr()),
+            program_counter: Delta(1.expr()),
+            stack_pointer: Delta(1.expr()),
+            ..Default::default()
+        };
+        let opcode = cb.query_cell();
+        let same_context = SameContextGadget::construct(
+            cb,
+            opcode,
```

Comment:
```suggestion
            cb.query_cell(),
```

Reply:
I tried but the borrow checker complained that the `ConstraintBuilder` is mutably borrwoed previously (the first argument of `SameContextGadget::construct`, do I miss something to make this work?

Reply:
hmm, I see. I didn't compile it. I'll take a look.